service/worker/domaindeprecation/workflow.go

```diff
@@ -73,12 +80,38 @@ func (w *domainDeprecator) DomainDeprecationWorkflow(ctx workflow.Context, domai
 	err = workflow.ExecuteActivity(
 		workflow.WithActivityOptions(ctx, activityOptions),
 		w.DeprecateDomainActivity,
-		domainName,
+		domainParams,
 	).Get(ctx, nil)
 	if err != nil {
 		return err
 	}
 
+	// Step 3: List open workflows of the domain
+	var workflowDetails []WorkflowDetails
+	err = workflow.ExecuteActivity(
+		workflow.WithActivityOptions(ctx, activityOptions),
```

Comment:
A domain can have millions of open workflows. Listing all of them might not fit into memory. Also 5 minutes start-to-close timeout will not be sufficient to terminate all the workflows. 
Consider moving your batch logic here inside the workflow. In a loop, call ListAndTerminateActivity which takes pagetoken and page size parameters. It then returns the next page token. 
I'd also make use of custom heartbeats inside ListAndTerminateActivity to keep the index of last processed item in the current batch.